.github/workflows/production-build.yaml

```diff
@@ -15,11 +24,20 @@ jobs:
         service_name: [backend, frontend, document-service, platform-service, prompt-service, worker, x2text-service]
 
     steps:
-      - name: Checkout code
+      - name: Checkout code for release
+        if: github.event_name == 'release'
         uses: actions/checkout@v2
         with:
           ref: ${{ github.event.release.tag_name }}
 
+
+      - name: Checkout code for branch
+        if: github.event_name != 'release'
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.inputs.branch }}
```

Comment:
@johnyrahul I don't think this is required

Reply:
done